pilot/cmd/istioctl/inject.go

```diff
@@ -58,6 +59,34 @@ func getMeshConfigFromConfigMap(kubeconfig string) (*meshconfig.MeshConfig, erro
 	return model.ApplyMeshConfigDefaults(yaml)
 }
 
+func getInjectConfigFromConfigMap(kubeconfig string) (string, error) {
+	_, client, err := kube.CreateInterface(kubeconfig)
+	if err != nil {
+		return "", err
+	}
+	config, err := client.CoreV1().ConfigMaps(istioNamespace).Get(injectConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		log.Infof("could not find valid configmap %q from namespace  %q: %v - "+
+			"Use --injectConfigFile or re-run kube-inject with `-i <istioSystemNamespace> and ensure istio-inject configmap exists",
+			injectConfigMapName, istioNamespace, err)
+		return "", nil
+	}
+	// values in the data are strings, while proto might use a
+	// different data type.  therefore, we have to get a value by a
+	// key
+	injectData, exists := config.Data[injectConfigMapKey]
+	if !exists {
+		return "", fmt.Errorf("missing configuration map key %q", injectConfigMapKey)
+	}
+	log.Debugf("read inject template from configmap.")
```

Comment:
I'm assuming this was left in for debug and would be removed before merge. We need to be careful with logging to `stdout` since users often redirect `kube-inject` to `kubectl`, e.g. `istioctl kube-inject -f file.yaml | kubectl apply -f -`

Reply:
Agreed about being careful.  My thinking was the log level and target would control this and not output anything normally.  So, I figured debug level logs were ok to leave.  I'm fine to yank it.  

On the logging topic, I couldn't get logs to output to stdout anyway (even with`--log_output_level` and `--log_target` set.  I think because the log.Configure() is only called in the create subcommand.

Reply:
updating the thread-- created a PR for the logger fix: #3025